# make some work matrices allocatable to bypass errors generated by -ftree-vectorize with gcc 12.x on zen4/zen5
# author: Bart Oldeman (Digital Research Alliance of Canada)
diff -ur flexiblas-3.2.1.orig/contributed/lapack-3.10.1/SRC/zgbtrf.f flexiblas-3.2.1/contributed/lapack-3.10.1/SRC/zgbtrf.f
--- flexiblas-3.2.1.orig/contributed/lapack-3.10.1/SRC/zgbtrf.f	2023-04-03 14:07:31.000000000 +0000
+++ flexiblas-3.2.1/contributed/lapack-3.10.1/SRC/zgbtrf.f	2025-09-24 15:29:42.138091646 +0000
@@ -169,8 +169,8 @@
       COMPLEX*16         TEMP
 *     ..
 *     .. Local Arrays ..
-      COMPLEX*16         WORK13( LDWORK, NBMAX ),
-     $                   WORK31( LDWORK, NBMAX )
+      COMPLEX*16, ALLOCATABLE :: WORK13( :, : ),
+     $                   WORK31( :, : )
 *     ..
 *     .. External Functions ..
       INTEGER            ILAENV, IZAMAX
@@ -232,6 +232,8 @@
 *
 *        Use blocked code
 *
+         ALLOCATE(WORK13(LDWORK, NBMAX), WORK31(LDWORK, NBMAX))
+*
 *        Zero the superdiagonal elements of the work array WORK13
 *
          DO 20 J = 1, NB
